manifests/init.pp

```diff
@@ -141,8 +145,14 @@
   Boolean $pulp_db_unsafe_autoretry = false,
   Optional[Enum['majority', 'all']] $pulp_db_write_concern = undef,
   Boolean $pulp_manage_db = true,
+  Boolean $pulp_expose_db = false,
+  Boolean $pulp_expose_db_ipv6 = false,
 ) {
 
+  if !$pulp_manage_db and $pulp_expose_db {
+    fail("pulp_expose_db cannot be set to true if pulp_manage_db is false")
+  }
```

Comment:
I think it makes more sense to have this in `pulp.pp`. It's compiled anyway so it fails regardless, but that way the standalone use of `katello::pulp` also benefits from the check. Design wise it's also better to keep all Pulp logic in Pulp itself. When dropping Pulp 2, we can simply remove pulp.pp and drop all unused parameters.